spec.html

```diff
@@ -4906,6 +4980,89 @@ <h1>CopyDataProperties ( _target_, _source_, _excludedItems_ )</h1>
       </emu-note>
     </emu-clause>
 
+    <emu-clause id="sec-newprivatename" aoid="NewPrivateName">
+      <h1>NewPrivateName ( _description_ )</h1>
+      <emu-alg>
+        1. Return a new unique Private Name whose [[Description]] value is _description_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-privatefieldfind" aoid="PrivateFieldFind">
+      <h1>PrivateFieldFind ( _P_, _O_ )</h1>
+      <emu-alg>
+        1. Assert: _P_ is a Private Name.
+        1. Assert: _O_ is an object with a [[PrivateFieldValues]] internal slot.
+        1. For each element _entry_ in _O_.[[PrivateFieldValues]],
+          1. If _entry_.[[PrivateName]] is _P_, return _entry_.
+        1. Return ~empty~.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-privatefieldadd" aoid="PrivateFieldAdd">
+      <h1>PrivateFieldAdd ( _P_, _O_, _value_ )</h1>
+      <emu-alg>
+        1. Assert: _P_ is a Private Name.
+        1. Assert: Type(_O_) is Object.
+        1. Let _entry_ be PrivateFieldFind(_P_, _O_).
+        1. If _entry_ is not ~empty~, throw a *TypeError* exception.
+        1. Append { [[PrivateName]]: P, [[PrivateFieldValue]]: _value_ } to _O_.[[PrivateFieldValues]].
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-privatefieldget" aoid="PrivateFieldGet">
+      <h1>PrivateFieldGet ( _P_, _O_ )</h1>
+      <emu-alg>
+        1. Assert: _P_ is a Private Name.
+        1. Assert: Type(_O_) is Object.
+        1. Let _entry_ be PrivateFieldFind(_P_, _O_).
+        1. If _entry_ is ~empty~, throw a *TypeError* exception.
+        1. Return _entry_.[[PrivateFieldValue]].
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-privatefieldset" aoid="PrivateFieldSet">
+      <h1>PrivateFieldSet ( _P_, _O_, _value_ )</h1>
+      <emu-alg>
+        1. Assert: _P_ is a Private Name.
+        1. Assert: Type(_O_) is Object.
+        1. Let _entry_ be PrivateFieldFind(_P_, _O_).
+        1. If _entry_ is ~empty~, throw a *TypeError* exception.
+        1. Set _entry_.[[PrivateFieldValue]] to _value_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-definefield" aoid="DefineField">
+      <h1>DefineField ( _receiver_, _fieldRecord_ )</h1>
+      <emu-alg>
+        1. Assert: Type(_receiver_) is Object.
+        1. Assert: _fieldRecord_ is a Record as created by ClassFieldDefinitionEvaluation.
```

Comment:
let's make a named record type, instead of tying this to a specific abstract operation.